drivers/can/can_mspm0_canfd.c

```diff
@@ -0,0 +1,276 @@
+/*
+ * Copyright (c) 2024 Texas Instruments
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/irq.h>
+
+#include <zephyr/drivers/can.h>
+#include <zephyr/drivers/can/can_mcan.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/clock_control/mspm0_clock_control.h>
+#include <zephyr/logging/log.h>
+
+/* Driverlib includes */
+#include <ti/driverlib/dl_mcan.h>
+
+LOG_MODULE_REGISTER(can_mspm0_canfd, CONFIG_CAN_LOG_LEVEL);
+
+#define DT_DRV_COMPAT ti_mspm0_canfd
+
+#define CAN_MSPM0_CLK_SRC_PLLCLK1 1
+
+struct can_mspm0_canfd_config {
+	uint32_t ti_canfd_base;
```

Comment:
You could store this as `MCAN_Regs *` instead and save the typecasts when setting registers.